src/tasks/FilterFeedTask.ts

```diff
@@ -4,40 +4,51 @@ import { TextChannel, Channel } from 'discord.js';
 import * as log4js from 'log4js';
 import Task from './Task';
 import { NewsUtil } from '../util/NewsUtil';
+import { JiraUtil } from '../util/JiraUtil';
 import MojiraBot from '../MojiraBot';
 
 export default class FilterFeedTask extends Task {
 	private static logger = log4js.getLogger( 'FilterFeedTask' );
 
 	private channel: Channel;
 	private jql: string;
+	private jqlRemoved: string;
 	private filterFeedEmoji: string;
 	private title: string;
 	private titleSingle: string;
 	private publish: boolean;
+	private cached: boolean;
 
 	private knownTickets = new Set<string>();
 
+	private	lastRun: string;
+
 	constructor( feedConfig: FilterFeedConfig, channel: Channel ) {
 		super();
 
 		this.channel = channel;
 		this.jql = feedConfig.jql;
+		this.jqlRemoved = feedConfig.jqlRemoved ?? '';
 		this.filterFeedEmoji = feedConfig.filterFeedEmoji;
 		this.title = feedConfig.title;
 		this.titleSingle = feedConfig.titleSingle || feedConfig.title.replace( /\{\{num\}\}/g, '1' );
 		this.publish = feedConfig.publish ?? false;
+		this.cached = feedConfig.cached ?? true;
```

Comment:
I don't mind refactoring it for this PR

Reply:
If I understand, you're saying to leave the variable, but to initiate a different task for the filter feed

Reply:
Basically what I'm saying is that this file should be split up into `VersionFeedTask.ts` and `CachedVersionFeedTask.ts`. Then in `MojiraBot.ts`, for each filter feed task to launch, it checks whether `feedConfig.cached` is true or not, and depending on that launches either a `VersionFeedTask` or a `CachedVersionFeedTask`.

The variable in the _config_ should stay (because I don't really want a separate config option for cached version feed tasks similar to how we have version feed tasks), but I'd like to get rid of the variable inside of the actual logic since it makes stuff messy.

I hope I could bring well enough across what I'm thinking of.

Reply:
Yep, I understand exactly what you're saying. I'll do this in a bit.

Reply:
Done